### PR TITLE
simple-list inputs can add items to a list via the store

### PR DESCRIPTION
**Pull request by amycheng**
closes https://github.com/clay/clay-kiln/issues/1083

### CODE REVIEW

**Comment by pfeytser** [comment]
@amycheng Is this ready to be code reviewed?

**Comment by amycheng** [comment]
@pfeytser yes!

File: inputs/simple-list.vue [hunk 1]
```diff
@@ -265,8 +266,42 @@
...
+            // validate that the list has items with these properties
+            stringProperty = getProp(items, 'text');
+            countProperty = getProp(items, 'count');
+
+            if (stringProperty && countProperty) {
+              itemIndex = getItemIndex(items, newItem.text, 'text');
+
+              if (itemIndex !== -1) {
+                // increase count if the item already exists in the list
+                items[itemIndex][countProperty] ++;
```
Comment (nelsonpecora) [inline inputs/simple-list.vue]: extra space before `++`

File: inputs/simple-list.vue [hunk 2]
```diff
@@ -265,8 +266,42 @@
...
+                return items;
+              } else {
+                // add item to the list
+                _.set(newItem, countProperty, 1);
+                return addListItem(items, newItem);
+              }
+            } else if (_.isString(_.head(items))) {
+              // if the list is just an array of strings, just add the string
+              // property
+              return addListItem(items, newItem['text']);
```
Comment (nelsonpecora) [inline inputs/simple-list.vue]: `newItem.text`...?

File: inputs/simple-list.vue [hunk 3]
```diff
@@ -265,8 +266,42 @@
...
+            } else if (items.length === 0) {
+              log.error('The list is empty, unable to determine data structure. ', { action: 'adding item to a list' });
```
Comment (nelsonpecora) [inline inputs/simple-list.vue]: extra space after period

File: inputs/simple-list.vue (hunk 2 above)
Comment (amycheng) [reply inputs/simple-list.vue]: lolo woops

**Pull request merged**